hathor/cli/side_dag.py

```diff
@@ -205,104 +195,82 @@ def is_option(arg_: str) -> bool:
     return hathor_node_argv, side_dag_argv
 
 
-def _run_side_dag_node(argv: list[str], *, hathor_node_process: Process, conn: 'Connection') -> None:
+def _run_monitor_process(*processes: Process) -> None:
     """Function to be called by the main process to run the side-dag full node."""
-    logger.info('waiting for hathor node to initialize...')
-    if not conn.poll(HATHOR_NODE_INIT_TIMEOUT):
-        logger.critical(
-            f'side-dag node not started because hathor node failed to initialize before {HATHOR_NODE_INIT_TIMEOUT} '
-            f'seconds timeout'
-        )
-        return
+    while True:
+        time.sleep(MONITOR_WAIT_PERIOD)
+        for process in processes:
+            if not process.is_alive():
+                return _kill_all_processes(processes, exited=process)
 
-    message = conn.recv()
-    if isinstance(message, HathorProcessInitFail):
-        logger.critical(f'side-dag node not started because hathor node initialization failed:\n{message.reason}')
-        return
 
-    assert isinstance(message, HathorProcessInitSuccess)
-    logger.info('hathor node initialized')
-    logger.info('starting side-dag node...')
+def _kill_all_processes(processes: tuple[Process, ...], *, exited: Process) -> None:
+    logger.critical(f'process "{exited.name}" exited, terminating all nodes')
+    for process in processes:
+        if process.is_alive():
+            logger.critical(f'terminating process "{process.name}"...')
+            process.terminate()
 
-    try:
-        side_dag = SideDagRunNode(argv=argv)
-    except (BaseException, Exception):
-        logger.exception('terminating hathor node due to exception in side-dag node...')
-        conn.send(SideDagProcessTerminated())
-        hathor_node_process.terminate()
-        return
-
-    side_dag.run()
-
-    # If `run()` returns, either the hathor node exited and terminated us, leaving the message below, or the side-dag
-    # node exited and we will terminate the hathor node.
-    if conn.poll():
-        message = conn.recv()
-        assert isinstance(message, HathorProcessTerminated)
-        logger.critical('side-dag node terminated because hathor node exited')
-        return
+    time.sleep(MONITOR_WAIT_PERIOD)
+    for process in processes:
+        if process.is_alive():
+            logger.critical(f'process "{process.name}" still alive, killing it...')
```

Comment:
Maybe log the pid too.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/1079/commits/e965a13f46986b636960070abbf4693bea9fc5c7

Reply:
Ok! Thanks!